storybookjs/storybook
# [documentation] Updates theming docs to use withOptions()

Provides more clear guidance on setting up theming using @storybook/addon-options, given that decorators require additional configuration and setOptions() is being deprecated in favor of withOptions().

Issue:

## What I did
Updated documentation related to setting up custom theming in Storybook 4.x by using withOptions() and decorators in `.storybook/config.js`.

## How to test

Is this testable with Jest or Chromatic screenshots?
No, it is markdown documentation.

Does this need a new example in the kitchen sink apps?
No.

Does this need an update to the documentation?
Yes (as it is an update to documentation).

If your answer is yes to any of these, please make sure to include it in your PR.

For maintainers only: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
